File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    
    limit = len(nums) / 2
    
    pockets = set()
    for num in nums:
        pockets.add(num)
    
    total_count = len(pockets)
    if len(pockets) >= limit:
        return limit
    else:
        return total_count
        

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int limit = nums.length / 2;
        
        Set<Integer> set = new HashSet<>();
        for (int i=0; i<nums.length; i++) {
            set.add(nums[i]);
        }
        
        if (set.size() < limit) {
            return set.size();
        }
        
        return limit;
        
    }
}